emukit/examples/gp_bayesian_optimization/random_search.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import numpy as np
+
+from emukit.core.loop.outer_loop import OuterLoop
+from emukit.bayesian_optimization.loops.bayesian_optimization_loop import BayesianOptimizationResults
+from emukit.core.loop.candidate_point_calculators import CandidatePointCalculator
+from emukit.experimental_design.model_free.random_design import RandomDesign
+from emukit.core.loop.loop_state import LoopState
+from emukit.core.parameter_space import ParameterSpace
+from emukit.core.loop.model_updaters import ModelUpdater
+from emukit.core.loop.loop_state import create_loop_state
+
+
+class RandomSampling(CandidatePointCalculator):
+
+    def __init__(self, parameter_space: ParameterSpace):
+        """
+        Samples a new candidate point uniformly at random
+
+        :param parameter_space: Input space
+        """
+        self.rd = RandomDesign(parameter_space=parameter_space)
+
+    def compute_next_points(self, loop_state: LoopState, context: dict=None) -> np.ndarray:
+        """
+        :param loop_state: Object that contains current state of the loop
+        :param context: Contains variables to fix through optimization of acquisition function. The dictionary key is
+                        the parameter name and the value is the value to fix the parameter to.
+        :return: (n_points x n_dims) array of next inputs to evaluate the function at
+        """
+        return self.rd.get_samples(1)
```

Comment:
This currently ignores the context. You could do:

```python
if context_manager is not None:
    sample = context_manager.context_free_space.sample_uniform(1)
    sample = context_manager.expand_vector(sample)
else:
    sample = self.parameter_space.sample_uniform(1)
return sample
```

Reply:
good point, I will add that